### PR TITLE
Fix/unlock after reassignment

### PR DESCRIPTION
**Pull request by miggol**
I've moved the assignment logic to review_utils so it might be hard to see what's changed. The gist is as follows:

`update_go` now unsets `date_end` if there are unmade decisions. This, combined with setting `review.COMMISSION` in `assign_reviewers()` means a review reverts back to being open when a reviewer is added. (#300)

### CODE REVIEW

File: reviews/utils/review_utils.py [hunk 1]
```diff
@@ -431,3 +430,45 @@ def discontinue_review(review):
...
+
+
+def assign_reviewers(review, list_of_users, route):
+    """Assigns or reassigns a list of reviewers to a review"""
+
+    # Set stage to commission if users are assigned
+    if len(list_of_users) > 0:
+        review.stage = review.COMMISSION
+    else:
+        review.stage = review.ASSIGNMENT
```
Comment (tymees) [inline reviews/utils/review_utils.py:442]: Does this actually happen?

The assignment form has this validation rule:
```python
    def clean_reviewers(self):
        reviewers = self.cleaned_data['reviewers']

        if len(reviewers) == 0:
            raise ValidationError(
                _('Er moet tenminste één beoordelaar geselecteerd worden.'),
                code='no_reviewer')

        # See PR #188 if a secretary should be added to every review
        return self.cleaned_data['reviewers']
```

File: reviews/utils/review_utils.py [hunk 2]
```diff
@@ -431,3 +430,45 @@ def discontinue_review(review):
...
+        review.date_should_end = timezone.now() + \
+            timezone.timedelta(
+                weeks=settings.SHORT_ROUTE_WEEKS
+            )
+    elif review.date_should_end is not None:
+        # We have no desired end date for long track reviews
+        review.date_should_end = None
+
+    # Create a new Decision for new reviewers
+    start_review_route(review, new_reviewers, route)
```
Comment (tymees) [inline reviews/utils/review_utils.py:460]: Like, not your fault, but this function name is really misguiding. ``create_decision`` would be a better idea

File: reviews/utils/review_utils.py (hunk 1 above)
Comment (miggol) [reply reviews/utils/review_utils.py:442]: Nope, this should never happen under normal circumstances. But when moving it into the utils package I took the liberty of adding this to make the function more utility-like, should we ever want to bump an application back to the assignment phase manually.

Comment (tymees) [reply reviews/utils/review_utils.py:442]: That makes sense, good thinking

File: reviews/utils/review_utils.py (hunk 2 above)
Comment (miggol) [reply reviews/utils/review_utils.py:460]: I know, I hate it too

**Pull request merged**